### PR TITLE
Properly package the app installer

### PR DESCRIPTION
**Pull request by smahmed776**

### CODE REVIEW

File: PYTHON/generate_manifest.py [hunk 1]
```diff
@@ -102,7 +102,9 @@ def browse_directories(dir_path: str, cur_type: Optional[str] = None):
             __generated_nodes.append(n_file_name)
         except Exception as e:
             print(
-                "❌ Failed to generate manifest from ",
+                "❌ Failed to generate manifest from ".encode("ascii", "replace").decode(
```
Comment (JeffDotPng) [inline PYTHON/generate_manifest.py:105]: Why does this need ascii encoding? Does UTF-8 break things? If it's the emojis breaking we could probably just remove them

File: electron/api/index.ts [hunk 2]
```diff
@@ -5,4 +5,18 @@ export default {
...
+        ipcRenderer.send(channel, data);
+      }
+    },
+    receive: (channel:string, func:(...arg:string[]) => void) => {
+      const validChannels = ['backend'];
+      if (validChannels.includes(channel)) {
+        // Deliberately strip event as it includes `sender`
+        ipcRenderer.on(channel, (event, ...args:string[]) => func(...args));
+      }
+    }
```
Comment (JeffDotPng) [inline electron/api/index.ts:21]: Why do this + defining `validChannels` when you could just wrap the channels like with `setUnsavedChanges`?

File: electron/main/backend.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
+import * as childProcess from "child_process";
+import { join, resolve } from "path";
+import treeKill from "tree-kill";
+
+const sendBackendLogToStudio = (win: Electron.BrowserWindow, data: string) => {
```
Comment (JeffDotPng) [inline electron/main/backend.ts:5]: `Electron.BrowserWindow` gives an error, need to `import { BrowserWindow } from 'electron'`

File: electron/main/backend.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+      sendBackendLogToStudio(win, data);
+      if (data.toString().includes(matchText)) {
+        sendBackendLogToStudio(win, "backend initialized successfully!");
+        resolve({ script });
+      }
+    });
+    script.addListener("exit", (code) => {
+      sendBackendLogToStudio(
+        win,
+        "Error: Failed to initialize backend try re lunching app!",
```
Comment (JeffDotPng) [inline electron/main/backend.ts:37]:
```suggestion
        "Error: Failed to initialize backend, please try relaunching the app!",
```

File: electron/main/backend.ts [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        resolve({ script });
+      }
+    });
+    script.stderr?.on("data", function (data) {
+      sendBackendLogToStudio(win, data);
+      if (data.toString().includes(matchText)) {
+        sendBackendLogToStudio(win, "backend initialized successfully!");
+        resolve({ script });
+      }
+    });
```
Comment (JeffDotPng) [inline electron/main/backend.ts:33]: The function name makes it seem like it's supposed to be a general function for running any command, but the logging from `win.webContents.send` show that it's only supposed to be used for backend initialization?

File: electron/main/backend.ts [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+import { join, resolve } from "path";
+import treeKill from "tree-kill";
+
+const sendBackendLogToStudio = (win: Electron.BrowserWindow, data: string) => {
+  if (global.initializingBackend) {
+    win.webContents.send("backend", data);
+  }
+};
+
+export const runCommand = (
```
Comment (JeffDotPng) [inline electron/main/backend.ts:11]: Can you use async/await instead of using `new Promise`?

**Comment by smahmed776** [comment]
Changes moved to #854

**Pull request closed without merging**